services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OrcidOAuth2Provider.java

```diff
@@ -87,12 +69,30 @@ public static String parseOrcidId(String body){
 		}
 	}
 
-
+	public static AliasAndType parseAliasAndType(Token accessToken) {
+		String orcid = parseOrcidId(accessToken.getRawResponse());
+		return new AliasAndType(convertOrcIdToURI(orcid), AliasType.USER_ORCID);
+	}
+	
 	@Override
 	public ProvidedUserInfo validateUserWithProvider(String authorizationCode, String redirectUrl) {
-		throw new IllegalArgumentException("This is not supported for ORCID.");
-	}
+		if (redirectUrl == null) {
+			throw new IllegalArgumentException("RedirectUrl cannot be null");
+		}
+		try {
+			OpenIdService service = (new OpenIdApi(authUrl, tokenUrl, userInfoUrl)).
+					createService(new OAuthConfig(apiKey, apiSecret, null, null, null, null));
 
+			Token accessToken = service.getAccessToken(null, new Verifier(authorizationCode));
```

Comment:
rename accessToken to accessTokenResponse (since the response contains multiple fields of which the access_token is just one).